Gordon360/Services/HousingService.cs

```diff
@@ -0,0 +1,774 @@
+                if (apartmentChoiceResult == null)
+                {
+                    throw new ResourceCreationException() { ExceptionMessage = "Apartment choice with ID " + applicationID + " and hall name " + apartmentChoice.HallName + " could not be inserted." };
+                }
+            }
+
+            // Update the info of apartment choices from the frontend that are already in the database
+            foreach (ApartmentChoiceViewModel apartmentChoice in apartmentChoicesToUpdate)
+            {
+                // All SqlParameters must be remade before being reused in an SQL Query to prevent errors
+                appIDParam = new SqlParameter("@APPLICATION_ID", applicationID);
+                rankingParam = new SqlParameter("@RANKING", apartmentChoice.HallRank);
+                buildingCodeParam = new SqlParameter("@BLDG_CDE", apartmentChoice.HallName);
+
+                int? apartmentChoiceResult = context.Database.ExecuteSqlCommand("UPDATE_AA_APARTMENT_CHOICE @APPLICATION_ID, @RANKING, @BLDG_CDE", appIDParam, rankingParam, buildingCodeParam); //run stored procedure
+                if (apartmentChoiceResult == null)
+                {
+                    throw new ResourceCreationException() { ExceptionMessage = "Apartment choice with ID " + applicationID + " and hall name " + apartmentChoice.HallName + " could not be updated." };
+                }
+            }
+
+            // Remove apartment choices from the database that were removed from the frontend
+            foreach (ApartmentChoiceViewModel apartmentChoice in apartmentChoicesToRemove)
+            {
+                // All SqlParameters must be remade before being reused in an SQL Query to prevent errors
+                appIDParam = new SqlParameter("@APPLICATION_ID", applicationID);
+                rankingParam = new SqlParameter("@RANKING", apartmentChoice.HallRank);
+                buildingCodeParam = new SqlParameter("@BLDG_CDE", apartmentChoice.HallName);
+
+                int? apartmentChoiceResult = context.Database.ExecuteSqlCommand("DELETE_AA_APARTMENT_CHOICE @APPLICATION_ID, @BLDG_CDE", appIDParam, buildingCodeParam); //run stored procedure
+                if (apartmentChoiceResult == null)
+                {
+                    throw new ResourceNotFoundException() { ExceptionMessage = "Apartment choice with ID " + applicationID + " and hall name " + apartmentChoice.HallName + " could not be removed." };
+                }
+            }
+
+            //--------
+            // Update the date modified (and application editor if necessary)
+
+            DateTime now = System.DateTime.Now;
+
+            appIDParam = new SqlParameter("@APPLICATION_ID", applicationID);
+            SqlParameter timeParam = new SqlParameter("@NOW", now);
```

Comment:
I don't think you need to save the current DateTime, since you only reference it once when creating the SqlParameter. Also, I believe the `System` namespace is unnecessary in this context.
```suggestion
            appIDParam = new SqlParameter("@APPLICATION_ID", applicationID);
            SqlParameter timeParam = new SqlParameter("@NOW", DateTime.Now);
```